### PR TITLE
Docs: add a script to deploy the combined Haddock to GH pages

### PR DESCRIPTION
**Pull request by michaelpj**
At the moment the only place we really have to host our docs is the GH pages site associated with this repo.

Automatically deploying to this site is a bit of a pain, so I've written a script to make at least the actual deployment part automated. We still need to run this script manually whenever we want to deploy the docs. (When do we do that? Who knows, we don't exactly have releases or anything.)

### CODE REVIEW

File: deploy-docs.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+set -eu
+
+GIT_ROOT=$(git rev-parse --show-toplevel)
+
+cd "$GIT_ROOT"
+
+if [ -z "$(git status --porcelain)" ]
```
Comment (j-mueller) [inline deploy-docs.sh]: The script doesn't get past this line for me, should it be `-n` instead of `-z` here?

File: deploy-docs.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+WORKTREE_NAME=$BRANCH
+
+echo "Clearing worktree"
+rm -rf $WORKTREE_NAME
+git worktree prune
+rm -rf .git/worktrees/$WORKTREE_NAME/
+mkdir $WORKTREE_NAME
+
+echo "Checking out $BRANCH branch into $WORKTREE_NAME"
+git worktree add -B $BRANCH $WORKTREE_NAME $UPSTREAM_BRANCH
```
Comment (j-mueller) [inline deploy-docs.sh]: For this to work you must have run `git fetch upstream gh-pages` at least once

File: deploy-docs.sh (hunk 1 above)
Comment (michaelpj) [reply deploy-docs.sh]: Damn, I switched away from `[[` because shellcheck dinged me, then I always get the tests wrong :(

**Comment by michaelpj** [comment]
Updated.

**Pull request merged**